taichi/transforms/demote_operations.cpp

```diff
@@ -14,6 +14,25 @@ class DemoteOperations : public BasicStmtVisitor {
   DemoteOperations() : BasicStmtVisitor() {
   }
 
+  void visit(BitExtractStmt *stmt) override {
+    // @ti.func
+    // def bit_extract(input, begin, end):
+    //   return (input >> begin) & ((1 << (end - begin)) - 1)
+    VecStatement statements;
+    auto begin = statements.push_back<ConstStmt>(
+        LaneAttribute<TypedConstant>(stmt->bit_begin));
```

Comment:
Also, the data type of this statement should be the same with `stmt->input`, to avoid additional type casts by the following `type_check`.

Reply:
Btw, is `DataType::f32` still there after #1905? What's the replacement?

Reply:
I think it becomes `PrimitiveType::f32` now. But I don't get the point why you need `f32`...